src/internal/operations/export_test.go

```diff
@@ -0,0 +1,267 @@
+
+			err = op.persistResults(ctx, now, &test.stats)
+			test.expectErr(t, err, clues.ToCore(err))
+
+			assert.Equal(t, test.expectStatus.String(), op.Status.String(), "status")
+			assert.Equal(t, len(test.stats.cs), op.Results.ItemsRead, "items read")
+			assert.Equal(t, test.stats.bytesRead.NumBytes, op.Results.BytesRead, "resource owners")
+			assert.Equal(t, test.stats.resourceCount, op.Results.ResourceOwners, "resource owners")
+			assert.Equal(t, now, op.Results.StartedAt, "started at")
+			assert.Less(t, now, op.Results.CompletedAt, "completed at")
+		})
+	}
+}
+
+type expCol struct {
+	base  string
+	items []export.Item
+}
+
+func (ec expCol) BasePath() string { return ec.base }
+func (ec expCol) Items(ctx context.Context) <-chan export.Item {
+	ch := make(chan export.Item)
+
+	go func() {
+		defer close(ch)
+
+		for _, item := range ec.items {
+			ch <- item
+		}
+	}()
+
+	return ch
+}
+
+func (suite *ExportOpSuite) TestZipExports() {
+	table := []struct {
+		name       string
+		collection []export.Collection
+		shouldErr  bool
+		readErr    bool
+	}{
+		{
+			name:       "nothing",
+			collection: []export.Collection{},
+			shouldErr:  true,
+		},
+		{
+			name: "empty",
+			collection: []export.Collection{
+				expCol{
+					base:  "",
+					items: []export.Item{},
+				},
+			},
+		},
+		{
+			name: "one item",
+			collection: []export.Collection{
+				expCol{
+					base: "",
+					items: []export.Item{
+						{
+							ID: "id1",
+							Data: export.ItemData{
+								Name: "test",
+								Body: io.NopCloser(bytes.NewBufferString("test")),
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "multiple items",
+			collection: []export.Collection{
+				expCol{
+					base: "",
+					items: []export.Item{
+						{
+							ID: "id1",
+							Data: export.ItemData{
+								Name: "test",
+								Body: io.NopCloser(bytes.NewBufferString("test")),
+							},
+						},
+					},
+				},
+				expCol{
+					base: "/fold",
+					items: []export.Item{
+						{
+							ID: "id2",
+							Data: export.ItemData{
+								Name: "test2",
+								Body: io.NopCloser(bytes.NewBufferString("test2")),
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "one item with err",
+			collection: []export.Collection{
+				expCol{
+					base: "",
+					items: []export.Item{
+						{
+							ID:    "id3",
+							Error: assert.AnError,
+						},
+					},
+				},
+			},
+			readErr: true,
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			zc, err := zipExportCollection(ctx, test.collection)
+
+			if test.shouldErr {
+				assert.Error(t, err, "error")
+				return
+			}
+
+			require.NoError(t, err, "error")
+
+			assert.Empty(t, zc.BasePath(), "base path")
+
+			count := 0
+			for item := range zc.Items(ctx) {
+				assert.Equal(t, "export.zip", item.Data.Name, "name")
+
+				_, err := io.Copy(io.Discard, item.Data.Body)
```

Comment:
how difficult would it be to actually check the items in the generated archive? Right now we don't have any automated tests that ensure the archive content is actually what we expect. Either a test here or something in the sanity tests or otherwise would help put my mind at ease

Reply:
Let me update the test to verify the data within the zip file.